src/models/position/facade.js

```diff
@@ -52,7 +52,15 @@ const prepareChangeParticipantPositionTransaction = async (transferList) => {
     const knex = await Db.getKnex()
     const participantName = transferList[0].value.content.payload.payerFsp
     const currencyId = transferList[0].value.content.payload.amount.currency
-    const settlementModel = await SettlementModelCached.getByLedgerAccountTypeId(Enum.Accounts.LedgerAccountType.POSITION)
+    const allSettlementModels = await SettlementModelCached.getAll()
+    let settlementModels = allSettlementModels.filter(model => model.currencyId === currencyId)
+    if (!(Array.isArray(settlementModels)) || settlementModels.length <= 0) {
+      settlementModels = allSettlementModels.filter(model => model.currencyId === null) // Default settlement model
+      if (!settlementModels || (!(Array.isArray(settlementModels) || settlementModels.length <= 0))) {
```

Comment:
is `Array.isArray` really required? If we're calling `.filter` in `58`, surely this is an array, right?

Reply:
I agree